app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterClaimController.scala

```diff
@@ -79,76 +78,86 @@ class EnterClaimController @Inject() (
             logger.warn("No duties found to create claims ")
             Redirect(routes.SelectDutiesController.selectDuties())
           case Right(Some(claims)) =>
-            val nextPage = calculateNextPage(claims)
+            val nextPage = calculateNextPage(claims, journeyBindable)
             updateClaimAnswer(claims, fillingOutClaim, nextPage)
         }
       }
     }
 
-  def enterClaim(id: UUID): Action[AnyContent] =
+  def enterClaim(taxCategory: TaxCategory, taxCode: TaxCode, journeyBindable: JourneyBindable): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswers[ClaimsAnswer] { (fillingOutClaim, answer) =>
-        answer.flatMap(_.find(_.id === id)) match {
+      implicit val journey: JourneyBindable = journeyBindable
+      withAnswersAndRoutes[ClaimsAnswer] { (_, answer, _) =>
+        answer.flatMap(_.find(claim => claim.taxCategory === taxCategory && claim.taxCode === taxCode)) match {
           case Some(claim) =>
-            fillingOutClaim.draftClaim.fold(_.isMrnFlow) match {
-              case true  =>
+            journeyBindable match {
+              case JourneyBindable.Single | JourneyBindable.Bulk =>
                 val emptyForm = mrnClaimAmountForm(claim.paidAmount)
                 val form      = Either.cond(claim.isFilled, emptyForm.fill(ClaimAmount(claim.claimAmount)), emptyForm).merge
-                Ok(enterClaimPage(id, form, claim))
-              case false =>
+                Ok(enterClaimPage(form, claim))
+              case JourneyBindable.Scheduled                     =>
                 val form = Either
                   .cond(
                     claim.isFilled,
-                    entryClaimAmountForm.fill(ClaimAndPaidAmount(claim.paidAmount, claim.claimAmount)),
-                    entryClaimAmountForm
+                    scheduledClaimAmountForm.fill(ClaimAndPaidAmount(claim.paidAmount, claim.claimAmount)),
+                    scheduledClaimAmountForm
                   )
                   .merge
-                Ok(enterEntryClaimPage(id, form, claim))
+                Ok(enterScheduledClaimPage(form, claim))
             }
           case None        =>
             Ok("This claim no longer exists") //TODO replace this with the a proper error page
```

Comment:
I've sent a message to the team to find out where we are with this error page. I understood it to have been designed.

Reply:
Thanks